torchrl/data/replay_buffers/samplers.py

```diff
@@ -475,6 +475,15 @@ def sample(self, storage: Storage, batch_size: int) -> torch.Tensor:
             index = index.unsqueeze(0)
         index.clamp_max_(len(storage) - 1)
         weight = torch.as_tensor(self._sum_tree[index])
```

Comment:
@vmoens it seems the `weight` here is not updated, I think we should do `weight = torch.as_tensor(self._sum_tree[index])` after changing the `index`?